app/scripts/views/sign_in_unblock.js

```diff
@@ -54,12 +54,15 @@ define(function (require, exports, module) {
     },
 
     onSignInError (account, password, err) {
+      const unblockCode = this.getElementValue('#unblock_code');
       if (AuthErrors.is(err, 'INCORRECT_PASSWORD')) {
         // The user must go enter the correct password this time.
         this.navigate(this._getAuthPage(), {
           email: account.get('email'),
           error: err
         });
+      } else if (AuthErrors.is(err, 'INVALID_UNBLOCK_CODE')) {
```

Comment:
This probably wasn't very apparent, but there are actually two validations that occur on the codes. `INVALID_UNBLOCK_CODE` is thrown when the client fails to validate the code, ex not correct length or type of characters. The other is `INCORRECT_UNBLOCK_CODE`, when the user submits the code and the server says it is invalid.

For this bug we want to use `INCORRECT_UNBLOCK_CODE`.

Reply:
Thanks for this explanation. :smiley: